app/modules/documents/documents.controller.js

```diff
@@ -121,21 +123,46 @@
 		 * @param  {object} doc Document object.
 		 */
 		function destroyDocument(doc) {
-			if (vm.documents.length > 1 && vm.currentDocument.id === doc.id) {
-				Document.destroy(doc.id)
-					.then(function() {
-						var last = vm.documents[vm.documents.length - 1];
-						vm.selectDocument(last);
-					});
+			if (vm.documents.length > 1 && vm.currentDocument && vm.currentDocument.id === doc.id) {
+				Document.destroy(doc.id).then(function() {
+					_selectLatestDocument();
+				});
+			} else if (vm.currentDocument && vm.currentDocument.id === doc.id) {
+				Document.destroy(doc.id).then(function() {
+					createDocument();
+				});
 			} else {
-				Document.destroy(doc.id)
-					.then(function() {
-						vm.createDocument();
-					});
+				Document.destroy(doc.id);
 			}
 		}
 
 
+		/* Private Methods */
+		function _selectLatestDocument() {
+			var params = {};
+
+			params.orderBy = [
+				['touched', 'ASC']
+			];
+
+			params.limit = 1;
+
+			vm.currentDocument = Document.filter(params)[0];
+		}
+
+
+		/**
+		 * Autosave when detecting changes to the ng-model
+		 *
+		 */
+
+		function autoSave() {
+			$scope.$watch("documents.currentDocument.content", function( newValue, oldValue ) {
+				updateDocument(vm.currentDocument);
+			});
+		}
+
```

Comment:
I would write the watcher like this:

$scope.$watch(function() {
    return vm.currentDocument.content;
}, function(newValue, oldValue) {
    // Nothing changed, no need to take action
    if(angular.equals(newValue, oldValue) {
        return;
    }
    updateDocument(vm.currentDocument);
});

I would also put the call to autoSave(); inside the activate() method - this way you have only 1 activation method to keep track of.